src/cobra/flux_analysis/deletion.py

```diff
@@ -405,3 +405,47 @@ def double_gene_deletion(
         processes=processes,
         **kwargs
     )
+
+
+@pd.api.extensions.register_dataframe_accessor("knockout")
+@pd.api.extensions.register_series_accessor("knockout")
+class KnockoutAccessor:
+    """Access unique combinations of reactions in deletion results."""
+
+    def __init__(self, pandas_obj):
+        """Set up the accessor.
+
+        Parameters:
+        -----------
+        pandas_obj : pd.DataFrame or pd.Series
+            A result from one of the deletion methods.
+        """
+        self._validate(pandas_obj)
+        self._result = pandas_obj
+
+    @staticmethod
+    def _validate(obj):
+        # verify it is a deletion results
+        if obj.index.name != "ids":
+            raise AttributeError("Must be DataFrame returned by a deletion method.")
+
+    def __getitem__(self, args):
+        """Return the deletion result for a particular set of knocked entities.
+
+        Parameters:
+        -----------
+        args : iterable of str, cobra.Reaction or cobra.Gene
+
+        Returns:
+        --------
+        pd.DataFrame
+            The deletion result where the chosen entities have been deleted.
+        """
+        if not any(isinstance(args, t) for t in [list, tuple, set]):
+            args = [args]
+
+        if isinstance(args[0], Reaction) or isinstance(args[0], Gene):
```

Comment:
I would not do this type checking here. Since iterables can be of very varied type (at least `frozenset` is an obvious missing one). I would try to access the `id` attribute and recover.

> It's easier to ask forgiveness than it is to get permission.

:wink:

Reply:
But this would allow any object with an id attribute which is to broad, isn't it?

Reply:
I think that's okay because within this context I would expect a user to only use cobrapy objects. And if a user attempts to pass metabolite IDs into a gene deletion result it seems okay to me that it will raise a `KeyError` or so.